schema/equals_test.go

```diff
@@ -25,29 +25,29 @@ import (
 	fuzz "github.com/google/gofuzz"
 )
 
-func (Schema) Generate(rand *rand.Rand, size int) reflect.Value {
-	s := Schema{}
+func (*Schema) Generate(rand *rand.Rand, size int) reflect.Value {
+	s := &Schema{}
 	f := fuzz.New().RandSource(rand).MaxDepth(4)
 	f.Fuzz(&s)
 	return reflect.ValueOf(s)
 }
 
-func (Map) Generate(rand *rand.Rand, size int) reflect.Value {
-	m := Map{}
+func (*Map) Generate(rand *rand.Rand, size int) reflect.Value {
+	m := &Map{}
 	f := fuzz.New().RandSource(rand).MaxDepth(4)
 	f.Fuzz(&m)
 	return reflect.ValueOf(m)
 }
 
-func (TypeDef) Generate(rand *rand.Rand, size int) reflect.Value {
-	td := TypeDef{}
+func (*TypeDef) Generate(rand *rand.Rand, size int) reflect.Value {
+	td := &TypeDef{}
 	f := fuzz.New().RandSource(rand).MaxDepth(4)
 	f.Fuzz(&td)
 	return reflect.ValueOf(td)
 }
 
-func (Atom) Generate(rand *rand.Rand, size int) reflect.Value {
-	a := Atom{}
+func (*Atom) Generate(rand *rand.Rand, size int) reflect.Value {
+	a := &Atom{}
```

Comment:
not sure why these are needed

Reply:
those seem to be a part of using quick.Check and are used to generate the values to check: https://golang.org/pkg/testing/quick/#Generator

Reply:
I understand, but I don't think you need to receive a pointer to Atom, especially since you don't use the atom.

Reply:
Without this change it gave me this:
```
--- FAIL: TestEquals (0.03s)
panic: value method sigs.k8s.io/structured-merge-diff/v3/schema.Atom.Generate called using nil *Atom pointer [recovered]
	panic: value method sigs.k8s.io/structured-merge-diff/v3/schema.Atom.Generate called using nil *Atom pointer
```

Reply:
Wait, it worked before, what changed?

Reply:
it's probably because you changed the function signatures, revert the function signatures